TraderJoe-Bots/Large-Swap/src/agent.spec.ts

```diff
@@ -0,0 +1,452 @@
+  [createAddress("0xab852"), createAddress("0xac147")],
+];
+const testPairs: string[] = [
+  create2Pair(testTokens[0][0], testTokens[0][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+  create2Pair(testTokens[1][0], testTokens[1][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+  create2Pair(testTokens[2][0], testTokens[2][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+  create2Pair(testTokens[3][0], testTokens[3][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+  create2Pair(testTokens[4][0], testTokens[4][1], mockNetworkManager.factory, mockNetworkManager.pairInitCodeHash),
+];
+// Format: [sender, amount0In, amount1In, amount0Out, amount1Out, to]
+const testCases: [string, BigNumber, BigNumber, BigNumber, BigNumber, string][] = [
+  [
+    createAddress("0xab123"),
+    BigNumber.from("10"),
+    BigNumber.from("25"),
+    BigNumber.from("75"),
+    BigNumber.from("325"), // Greater than 150 threshold
+    createAddress("0xab321"),
+  ],
+  [
+    createAddress("0xac456"),
+    BigNumber.from("225"), // Greater than 200 threshold
+    BigNumber.from("25"),
+    BigNumber.from("75"),
+    BigNumber.from("15"),
+    createAddress("0xac789"),
+  ],
+  [
+    createAddress("0xad654"),
+    BigNumber.from("35"),
+    BigNumber.from("25"),
+    BigNumber.from("444"), // Greater than 300 threshold
+    BigNumber.from("15"),
+    createAddress("0xad987"),
+  ],
+  [
+    createAddress("0xad654"),
+    BigNumber.from("35"),
+    BigNumber.from("600"), // Greater than 350 threshold
+    BigNumber.from("45"),
+    BigNumber.from("15"),
+    createAddress("0xad987"),
+  ],
+  [
+    createAddress("0xae741"),
+    BigNumber.from("35"), // All lower than either
+    BigNumber.from("25"), // 400 or 450 thresholds
+    BigNumber.from("40"),
+    BigNumber.from("15"),
+    createAddress("0xae842"),
+  ],
+];
+// Format: [reserve0, reserve1, blockTimestampLast]
+const testReserves: [BigNumber, BigNumber, number][] = [
+  [BigNumber.from("1000"), BigNumber.from("1500"), 456],
+  [BigNumber.from("2000"), BigNumber.from("2500"), 891],
+  [BigNumber.from("3000"), BigNumber.from("3500"), 765],
+  [BigNumber.from("4000"), BigNumber.from("4500"), 852],
+  [BigNumber.from("5000"), BigNumber.from("5000"), 753],
+];
+const testBlocks: number[] = [444, 3230, 90059, 90210, 230608];
+
+describe("Large Swap test suite", () => {
+  let handleTransaction: HandleTransaction = provideHandleTransaction(
+    mockProvider as any,
+    mockNetworkManager,
+    testThresholdPercentage
+  );
+
+  const createTryAggregateCall = (
+    multicall2Address: string,
+    requireSuccess: boolean,
+    calls: [string, string][],
+    returnData: [boolean, string][],
+    blockNumber: number
+  ) => {
+    mockProvider.addCallTo(multicall2Address, blockNumber, MULTICALL2_IFACE, "tryAggregate", {
+      inputs: [requireSuccess, calls],
+      outputs: [returnData],
+    });
+  };
```

Comment:
I think it's fine to keep it the way it is in this case, but there's a way to test multicalls that's currently being used in UMEE #7 that might be easier (this is for `aggregate`, but adapting it to `tryAggregate` is straightforward):

https://github.com/NethermindEth/Forta-Agents/blob/845793101ba001a12acf075ee88f6b4a3ab79f55/Umee-Bots/High-Utilization-Ratio/src/agent.spec.ts#L46-L66

https://github.com/NethermindEth/Forta-Agents/blob/845793101ba001a12acf075ee88f6b4a3ab79f55/Umee-Bots/High-Utilization-Ratio/src/agent.spec.ts#L154-L159

This basically replaces the mock `call` from `mockProvider`, adding a sort of a proxy that decodes and makes the multicall calls one by one using `mockMulticallProvider` and encodes the results if the original call was made to the multicall address, or just redirects the call to `mockProvider`'s `call` otherwise.

Then, instead of having to specify every multicall input and output, you can just specify the multicall call results individually using, for example, `mockMulticallProvider.addCallTo`.